lib/nerves_hub_link/configurators/cert_key.ex

```diff
@@ -23,28 +20,11 @@ defmodule NervesHubLink.Configurator.Default do
   end
 
   defp maybe_add_cert(socket_opts) do
-    if socket_opts[:cert] || socket_opts[:certfile] do
-      # option already provided
-      socket_opts
-    else
-      cert =
-        Nerves.Runtime.KV.get(@cert)
```

Comment:
Should we keep this for backwards compatibility? I guess I don't know how much this is used. We can remove the mechanism to try and help put this there, but anyone who has already provisioned through the UBoot vars will be broken

Reply:
I guess we could, but maybe add logger warnings that this feature of provisioning is no longer supported in link?

I do like the concept of what provisioning does and how it works, but its current setup is a little brittle.

It would be nice if any `NERVES_` env var found during firmware burning was then converted to a uboot provisioning line. That would make it more flexible and also allow for us to remove the `provisioning.conf` file, etc.

Reply:
This would also allow for the shared secret auth configs to be stored in env vars and outside of the config, which I really like.